### PR TITLE
Add workflows for multiple OS and python versions

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
In this PR, I'm taking into account issue #186. 
I have added python versions and the missing operative systems. 
Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: .github/workflows/selfhosted.yml [hunk 1]
```diff
@@ -29,4 +29,4 @@ jobs:
         pip install .[tiiq]
     - name: Test tii5q
       run: |
-        srun -p qpu5q pytest src/qibolab --pyargs qibolab --skip-no-qpu --platforms tii5q
+        srun -p tii5q pytest src/qibolab --pyargs qibolab --skip-no-qpu --platforms tii5q
```
Comment (andrea-pasquale) [inline .github/workflows/selfhosted.yml:32]: I think that this is outdated at this point given the new qpus available in the cluster.

**Review by andrea-pasquale — COMMENTED** [review]
Thanks @Edoardo-Pedicillo.
Looks good to me.
@scarrazza @stavros11 since in this PR we are modifying the workflow should we rebase this PR to #199?

File: .github/workflows/selfhosted.yml (hunk 1 above)
Comment (Edoardo-Pedicillo) [reply .github/workflows/selfhosted.yml:32]: Now we have different qpus with 5 qubits, which one should be used for the tests?

Comment (andrea-pasquale) [reply .github/workflows/selfhosted.yml:32]: In main there is a different `selfhosted.yml` after we merged #235 and #218.
We don't specify anymore the platform in the yml file but the workflow is launched after putting the correct label in the pull request.
I think that you should just merge main here and fix possible conflicts.

**Comment by Edoardo-Pedicillo** [comment]
Thanks @andrea-pasquale, I have merged the`main` into this branch, and the workflows work well.

**Review by andrea-pasquale — APPROVED** [review]
Thanks @Edoardo-Pedicillo.
To me this PR is ready to merge.

**Comment by andrea-pasquale** [comment]
@scarrazza this PR is ready to merge. If you agree I will merge it now.

**Review by scarrazza — APPROVED** [review]
LGTM, please go ahead.

**Pull request merged**